8.0.0 fails to npm install

This version has a `postinstall` script to run simple-git-hooks, but this package is in the dev dependencies and honestly git hooks should not be installed for users. 

I personally also use simple-git-hooks but as a manual install step for developers.

```bash
.../node_modules/@metamask/key-tree postinstall$ simple-git-hooks
.../node_modules/@metamask/key-tree postinstall: sh: 1: simple-git-hooks: not found
 ELIFECYCLE  Command failed.
.../node_modules/simple-git-hooks postinstall$ node ./postinstall.js
Error: Process completed with exit code 1.
```

--- a/package.json
+++ b/package.json
@@ -29,7 +29,6 @@
     "build:source": "yarn build:esm && yarn build:cjs",
     "build:types": "tsc --project tsconfig.build.json",
     "generate-vectors": "ts-node scripts/generate-vectors.ts > ./test/vectors/derivation.json",
-    "postinstall": "simple-git-hooks",
     "lint": "yarn lint:eslint && yarn lint:constraints && yarn lint:misc --check && yarn lint:dependencies --check && yarn lint:changelog",
     "lint:changelog": "auto-changelog validate",
     "lint:constraints": "yarn constraints",
@@ -40,17 +39,6 @@
     "prepack": "./scripts/prepack.sh",
     "test": "jest && jest-it-up",
     "test:watch": "jest --watch"
-  },
-  "simple-git-hooks": {
-    "pre-commit": "yarn lint-staged && yarn dedupe --check"
-  },
-  "lint-staged": {
-    "*.{js,ts}": [
-      "eslint --fix"
-    ],
-    "!(CHANGELOG).{json,yml,md}": [
-      "prettier --write"
-    ]
   },
   "dependencies": {
     "@metamask/scure-bip39": "^2.1.0",
@@ -84,11 +72,9 @@
     "eslint-plugin-prettier": "^4.2.1",
     "jest": "^28.1.3",
     "jest-it-up": "^2.0.2",
-    "lint-staged": "^13.0.3",
     "prettier": "^2.7.1",
     "prettier-plugin-packagejson": "^2.3.0",
     "rimraf": "^3.0.2",
-    "simple-git-hooks": "^2.8.0",
     "ts-jest": "^28.0.7",
     "ts-node": "^10.9.1",
     "typedoc": "^0.23.19",
@@ -105,8 +91,6 @@
   "lavamoat": {
     "allowScripts": {
       "@lavamoat/preinstall-always-fail": false,
-      "$root$": false,
-      "simple-git-hooks": false,
       "@swc/core": true
     }
   }

